Update smtp.markdown (#18466)
home-assistant/home-assistant.github.io: source/_integrations/smtp.markdown

notify:
  - name: "NOTIFIER_NAME"
    platform: smtp
    server: "smtp.example.com"
    username: "YOUR_USER_NAME"

>     password: "YOUR_PASSWORD"

Reviewer: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample.

Author: I was divided in this one, on one hand, we have the pattern of not including optional configurations in the configuration example, and on the other, we have that the most common use case for SMTP integrations is to use an external SMTP provider that will usually require a username and password.

Given that we are including provider-specific configuration in the end I will not strongly argue about keeping those optional Configuration variables, so I am good about removing them if I could not convince you @frenck :)

Reviewer: There is no discussion on this part, this is following our documentation standards. Please remove those optional parameters. Thanks 👍

Author: Consider it done! I will revert to the original example, below, as the server is also an optional variable.
`# Example configuration.yaml entry
notify:
  - name: "NOTIFIER_NAME"
    platform: smtp
    sender: "YOUR_SENDER"
    recipient: "YOUR_RECIPIENT"`

    sender: "YOUR_SENDER"
    recipient: "YOUR_RECIPIENT"
```

> Check your e-mail provider configuration or help pages to get the correct SMTP settings.

Reviewer: ```suggestion

Check your e-mail provider configuration or help pages to get the correct SMTP settings.
```

notify:
  - name: "NOTIFIER_NAME"
    platform: smtp

>     server: "smtp.gmail.com"

Reviewer: Please use `example.com` is it designed/documented to use for examples by RFC2606

Author: I will do that and reword the text that comes before the example to make it clear that this is not a google mail example but a generic SMTP example.

    sender_name: "My Home Assistant"
```

Keep in mind that Google has some extra layers of protection which need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails, and be aware that Google will periodically turn it off if it is not used (no e-mail is sent).

> To avoid having your e-mail notifications broken due to the less secure apps behaviour, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.

Reviewer: This is very specific content that only applies to a specific email provider. Lets move this down in the documentation in a separate chapter, or even completely remove it.

Author: I agree........I just didn't do that as I thought it was up for some reason. I will keep in the top a generic SMTP configuration example and create a chapter at the end of the page for specific e-mail provider configuration examples.

              </html>
```

To learn more about how to use notifications in your automations, please see the [getting started with automation page](/getting-started/automation/).

> ## Specific E-Mail Provider Configuration
> Check below some configurations examples for specific e-mail providers. 
> If you are in doubt about the SMTP settings required, check your e-mail provider configuration or help pages for more information about its specific SMTP configuration. 

Reviewer: ```suggestion
## Specific E-Mail Provider Configuration

Check below some configurations examples for specific e-mail providers. 
If you are in doubt about the SMTP settings required, check your e-mail provider configuration or help pages for more information about its specific SMTP configuration. 
```

### Google Mail

A sample configuration entry for Google Mail.

```yaml
# Example configuration.yaml entry for Google Mail.
notify:
  - name: "NOTIFIER_NAME"
    platform: smtp
    server: "smtp.gmail.com"
    port: 587
    timeout: 15
    sender: "YOUR_USERNAME@gmail.com"
    encryption: starttls
    username: "YOUR_USERNAME@gmail.com"
    password: "YOUR_PASSWORD"
    recipient:
      - "RECIPIENT_1@example.com"
      - "RECIPIENT_N@example.com"
    sender_name: "SENDER_NAME"
```

> Keep in mind that Google has some extra layers of protection which need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails, and be aware that Google will periodically turn it off if it is not used (no e-mail is sent).

Reviewer: ```suggestion

Keep in mind that Google has some extra layers of protection that need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails. Be aware that Google will periodically turn it off if it is not used (no e-mail is sent).
```

> To avoid having your e-mail notifications broken due to the less secure apps behaviour, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.

Reviewer: ```suggestion
To avoid having your e-mail notifications broken due to the less secure app's behavior, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.
```
